### PR TITLE
S2A decrypt trailing zeros test

### PR DESCRIPTION
**Pull request by matthewstevenson88**
This PR adds an additional test case to the S2A record protoco, namely a test that decrypts a TLS 1.3 record that was constructed with padding-by-zeros after the ciphertext.

(Also, when I ran `clang_format_code`, it modified a few other files. Feel free to ignore these, as there are no functional changes.)

### CODE REVIEW

File: test/core/tsi/s2a/s2a_test_data.h [hunk 1]
```diff
@@ -31,6 +31,20 @@ std::vector<uint8_t> test_message_2 = {8};
 std::vector<uint8_t> test_message_3 = {46,  98, 101, 255, 213, 156, 15,  100,
                                        126, 45, 130, 239, 209, 13,  156, 89};
 
+std::vector<uint8_t> padded_zeros_message = {'1', '2', '3', '4', '5', '6'};
```
Comment (cesarghali) [inline test/core/tsi/s2a/s2a_test_data.h]: Should these be zeroes? It could be that the name is a bit confusing.

Comment (matthewstevenson88) [reply test/core/tsi/s2a/s2a_test_data.h]: No, the message is correct. I could change it to `message_encrypted_with_padded_zeros`?

Comment (cesarghali) [reply test/core/tsi/s2a/s2a_test_data.h]: Yes that would be clearer. Thanks!

Comment (matthewstevenson88) [reply test/core/tsi/s2a/s2a_test_data.h]: Done.

**Comment by matthewstevenson88** [comment]
Squashing and merging.

**Pull request merged**